netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -133,44 +136,86 @@ private SslHandler sslHandler(ChannelHandlerContext ctx) {
    * be negotiated, the {@code handler} is added and writes to the {@link io.netty.channel.Channel}
    * may happen immediately, even before the TLS Handshake is complete.
    */
-  public static ProtocolNegotiator tls(final SslContext sslContext,
-                                       String authority) {
+  public static ProtocolNegotiator tls(SslContext sslContext, String authority) {
     Preconditions.checkNotNull(sslContext, "sslContext");
-    final URI uri = GrpcUtil.authorityToUri(Preconditions.checkNotNull(authority, "authority"));
+    URI uri = GrpcUtil.authorityToUri(Preconditions.checkNotNull(authority, "authority"));
+    return tlsInternal(sslContext, uri.getHost(), uri.getPort());
+  }
 
-    return new ProtocolNegotiator() {
-      @Override
-      public Handler newHandler(Http2ConnectionHandler handler) {
-        ChannelHandler sslBootstrap = new ChannelHandlerAdapter() {
-          @Override
-          public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
-            SSLEngine sslEngine = sslContext.newEngine(ctx.alloc(), uri.getHost(), uri.getPort());
```

Comment:
I would have expected this would be enough:
```
String host = uri.getHost();
if (host == null) {
  host = authority;
}
SSLEngine sslEngine = sslContext.newEngine(ctx.alloc(), host, uri.getPort());
```

I'm not aware of anything using the port, and I just tested and it works fine with -1 as the port. (Although -1 isn't defined special meaning in `SslContextnewEngine()` or `SSLContext.createSSLEngine()`, `SSLEngine.getPeerPort()` defines -1 as no port known.)